src/setup_node_env/node_version_validator.js

```diff
@@ -37,16 +37,14 @@ var pkg = require('../../package.json');
 var currentVersion = (process && process.version) || null;
 var rawRequiredVersion = (pkg && pkg.engines && pkg.engines.node) || null;
 var requiredVersion = rawRequiredVersion ? 'v' + rawRequiredVersion : rawRequiredVersion;
-var isVersionValid = !!currentVersion && !!requiredVersion && currentVersion === requiredVersion;
+var requiredVersionMajorMinor = requiredVersion.match(/^v(\d+\.\d+)/)[1];
+var isVersionValid = requiredVersionMajorMinor === currentVersion.match(/^v(\d+\.\d+)/)[1];
 
 // Validates current the NodeJS version compatibility when OpenSearch Dashboards starts.
 if (!isVersionValid) {
   var errorMessage =
-    'OpenSearch Dashboards does not support the current Node.js version ' +
-    currentVersion +
-    '. Please use Node.js ' +
-    requiredVersion +
-    '.';
+    `OpenSearch Dashboards was built with ${requiredVersion} and does not support the current Node.js version ${currentVersion}. ` +
+    `Please use Node.js ~v${requiredVersionMajorMinor} or higher.`;
```

Comment:
"Or higher" is currently not correct. Either it has to say `Please use Node.js ~v14.18.` or we actually block patch versions that are lower. Thoughts?

Reply:
Since the patch versions typically include fixes for CVEs, in the interest of security I think it would be best to block patch versions that are lower.